src/hotspot/share/runtime/handshake.cpp

```diff
@@ -373,39 +335,61 @@ bool Handshake::execute_direct(HandshakeClosure* thread_cl, JavaThread* target)
   // by the Handshakee.
   OrderAccess::acquire();
 
-  log_handshake_info(start_time_ns, op.name(), 1,  (pr == HandshakeState::_success) ? 1 : 0);
+  log_handshake_info(start_time_ns, op.name(), 1, executed_by_driver);
+}
+
+void Handshake::execute(AsynchHandshakeClosure* hs_cl, JavaThread* target) {
+  jlong start_time_ns = os::javaTimeNanos();
+  AsyncHandshakeOperation* op = new AsyncHandshakeOperation(hs_cl, target, start_time_ns);
 
-  return op.executed();
+  ThreadsListHandle tlh;
+  if (tlh.includes(target)) {
+    target->handshake_state()->add_operation(op);
+  } else {
+    log_handshake_info(start_time_ns, op->name(), 0, 0, "(thread dead)");
```

Comment:
It might be useful to also log the 'target' thread value here so:

    .... (thread=<ptr-value> is dead)"

Might be something like this:

log_handshake_info(start_time_ns, op->name(), 0, 0, "(thread=" INTPTR_FORMAT " is dead)", p2i(target));

Although you'd probably have to form the string in a buffer and then pass it
to the log_handshake_info() call... sigh...

Reply:
Fixed (via buffert)